linkerd/app/core/src/serve.rs

```diff
@@ -52,7 +52,7 @@ where
                                         .await
                                     {
                                         Ok(()) => debug!("Connection closed"),
-                                        Err(error) => info!(%error, "Connection closed"),
+                                        Err(reason) => debug!(%reason, "Connection closed"),
```

Comment:
something that might be worth doing (but we could do later) is looking at the error's reason and seeing if it was a normal connection close or some kind of local error, which might be an actual bug (e.g. if we get, like, EBADF, that's actually something bad).

then, we could keep the log event at debug, but make the message something like "Connection closed normally" if it was a not connected/connection reset/refused. that way, if people turn on the DEBUG level to try and diagnose some problem they're seeing, they see messages like "Connection closed normally" and it's less scary?

just a thought — we should merge this as-is in any case.